File: simexplorer-ide/simexplorer-ide-ant/org.openmole.ui/src/org/openmole/ui/palette/TaskNode.java
/*
 *  Copyright (C) 2010 Mathieu Leclaire <mathieu.leclaire@openmole.fr>
 * 
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 * 
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 * 
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package org.openmole.ui.palette;

import java.awt.datatransfer.DataFlavor;
import java.awt.datatransfer.Transferable;
import java.awt.datatransfer.UnsupportedFlavorException;
import java.io.IOException;
import org.openide.nodes.AbstractNode;
import org.openide.nodes.Children;
import org.openide.util.datatransfer.ExTransferable;

import org.openide.util.lookup.Lookups;
import org.openmole.core.workflow.model.task.IGenericTask;
import org.openmole.ui.commons.ApplicationCustomize;
import org.openmole.ui.workflow.implementation.Preferences;

/**
 *
 * @author Mathieu Leclaire <mathieu.leclaire@openmole.fr>
 */
public class TaskNode extends AbstractNode {

    private DataFlavor dataFlavor;
   // private Class<? extends IGenericTaskModelUI> model;
    private Class<? extends IGenericTask> coreTask;

    public TaskNode(DataFlavor key,
                  //  Class<? extends IGenericTaskModelUI> taskModel) {
                    Class<? extends IGenericTask> coreTask) {
        super(Children.LEAF, Lookups.fixed(new Object[]{key}));
        this.dataFlavor = key;
       // this.model = taskModel;
        this.coreTask = coreTask;

        setIconBaseWithExtension(Preferences.getInstance().getModelSettings(Preferences.getInstance().getModelClass(coreTask)).getThumbImagePath());
    }

     //DND start
    @Override
    public Transferable drag() throws IOException {
        ExTransferable retValue = ExTransferable.create( super.drag() );
        //add the 'data' into the Transferable
        retValue.put( new ExTransferable.Single(ApplicationCustomize.TASK_DATA_FLAVOR) {
            @Override
            protected Object getData() throws IOException, UnsupportedFlavorException 
            {return coreTask;}
            
        });
        return retValue;
    }
    //DND end
}

File: platform/gui/src/main/scala/org/openmole/ui/ide/palette/TaskNode.scala
/*
 * Copyright (C) 2011 Mathieu leclaire <mathieu.leclaire at openmole.org>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package org.openmole.ui.ide.palette

import java.awt.datatransfer.DataFlavor
import org.openmole.ui.ide.workflow.implementation.Preferences
import org.openmole.ui.ide.workflow.model.IEntityUI
import org.openmole.ui.ide.workflow.implementation.PropertyManager

class TaskNode(key: DataFlavor, task: IEntityUI) extends GenericNode(key,Preferences.properties(MoleConcepts.TASK_INSTANCE, task.entityType).getProperty(PropertyManager.THUMB_IMG),task.name){

}
//package org.openmole.ui.ide.palette;
//
//import java.awt.datatransfer.DataFlavor;
//import java.awt.datatransfer.Transferable;
//import java.awt.datatransfer.UnsupportedFlavorException;
//import java.io.IOException;
//import org.openide.util.datatransfer.ExTransferable;
//import org.openmole.commons.exception.UserBadDataError;
//
//import org.openmole.core.model.task.IGenericTask;
//import org.openmole.ui.ide.commons.ApplicationCustomize;
//import org.openmole.ui.ide.palette.Category.CategoryName;
//import org.openmole.ui.ide.workflow.implementation.IEntityUI;
//import org.openmole.ui.ide.workflow.implementation.Preferences;
//import org.openmole.ui.ide.workflow.implementation.PropertyManager;
//
///**
// *
// * @author Mathieu Leclaire <mathieu.leclaire@openmole.fr>
// */
//public class TaskNode extends GenericNode {
//
//    private IEntityUI task;
//
//    public TaskNode(DataFlavor key,
//                    IEntityUI task) throws UserBadDataError {
//        super(key,
//                Preferences.getInstance().getProperties(CategoryName.TASK_INSTANCE, task.getType()).getProperty(PropertyManager.THUMB_IMG),
//                task.getName());
//        this.task = task;
//    }
//
//    @Override
//    public Transferable drag() throws IOException {
//        ExTransferable retValue = ExTransferable.create( super.drag() );
//        retValue.put( new ExTransferable.Single(ApplicationCustomize.TASK_DATA_FLAVOR) {
//            @Override
//            protected Object getData() throws IOException, UnsupportedFlavorException 
//            {return task;}
//            
//        });
//        return retValue;
//    }
//}